modules/userId/eids.js

```diff
@@ -420,16 +420,26 @@ export function createEidsArray(bidRequestUserId) {
         eids = eids.concat(bidRequestUserId['pubProvidedId']);
       } else if (subModuleKey === 'ftrackId') {
         // ftrack has multiple IDs so we add each one that exists
-        let eid = {
+        const E_ID = {
           'atype': 1,
-          'id': (bidRequestUserId.ftrackId.DeviceID || []).join('|'),
           'ext': {}
+        };
+        const IDS = bidRequestUserId.ftrackId.hasOwnProperty('ext') ? bidRequestUserId.ftrackId.ext : bidRequestUserId.ftrackId;
+
+        for (let id in IDS) {
```

Comment:
To whom it may review.

1. The original logic expected `bidRequestUserId.ftrackId` to be the following schema.
```
ftrackId: {
  HHID: ['household_test_id'],
  DeviceID: ['device_test_id'],
  SingleDeviceID: ['single_device_test_id']
}
```
2. But during testing, we actually were seeing:
```
ftrackId: {
  uid: 'device_test_id',
  ext: {
    HHID: 'household_test_id',
    DeviceID: 'device_test_id',
    SingleDeviceID: 'single_device_test_id'
  }
}
```
This code change allows both JSON schema's to work correctly, but its not clear to me if both are needed. Do we know why the schema's are different and/or if one is incorrect?

Thanks!

Reply:
The ftrack ID module is what wraps the result in `ext`:

https://github.com/prebid/Prebid.js/blob/d8f25b92ed2d04ab578681076a35217548fc5df3/modules/ftrackIdSystem.js#L59-L64
https://github.com/prebid/Prebid.js/pull/8678

To me that looks odd, maybe unintentional - it'd make more sense to return `{ftrack: {hhid, deviceId, singleDeviceId}}`; but in the end what matters is that the callers (including the eid conversion logic) know what to expect.

Reply:
then I think you can simplify this to `E_ID.ext = bidRequestUserId.ftrackId.ext`; I don't see a need to handle other cases.

Reply:
Thanks @dgirardi , I concur. Our decode method is already doing that work so its def redundant at this step. I'll make those changes and ping when they are up. Thanks